dbms/src/Storages/MergeTree/ReplicatedMergeTreeQuorumWriter.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <unordered_map>
+#include <IO/ReadBuffer.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/ReadHelpers.h>
+
+#include <Storages/MergeTree/MergeTreeDataPart.h>
+
+namespace DB
+{
+
+struct ReplicatedMergeTreeQuorumWriter
+{
+    using PartitionIdToMaxBlock = std::unordered_map<String, Int64>;
+    using PartitonIdToPartName= std::unordered_map<String, String>;
+
+    PartitonIdToPartName added_parts;
+
+    MergeTreeDataFormatVersion format_version;
+
+    ReplicatedMergeTreeQuorumWriter(const std::string & old_added_parts, MergeTreeDataFormatVersion format_version_)
```

Comment:
Why the `old_` prefix? When we read from a table there are no old or new `added_parts`.